File: 3095. Shortest Subarray With OR at Least K I.py
class Solution:
    def minimumSubarrayLength(self, nums: List[int], k: int) -> int:
        answer = inf
        for start in range(len(nums)):
            current = 0
            for end in range(start, len(nums)):
                current = current | nums[end]
                if end - start + 1 < answer and current >= k:
                    answer = end - start + 1
        return -1 if answer == inf else answer

File: 3095. Shortest Subarray With OR at Least K I.go
func minimumSubarrayLength(nums []int, k int) int {
	answer := math.MaxInt32
	for start := range(nums) {
		current := 0
		for end := start; end < len(nums); end++ {
			current = current | nums[end]
			if current >= k && end - start + 1 < answer {
				answer = end - start + 1
			}
		}
	}
	if answer == math.MaxInt32 {
		return -1
	}
	return answer
}